### PR TITLE
[DPM-647] fix duplicate events

### PR DESCRIPTION
**Pull request by algys**
Added offset field for game session updates. And unique constraint by offset field, so now it doesn't allow duplicates.
P.S. it's not very good decision in terms of performance, but for now it's the only possible way...

### CODE REVIEW

File: eventprocessor/handlers.go [hunk 1]
```diff
@@ -262,11 +296,16 @@ func onGameMessage(ctx context.Context, p *EventProcessor, event *eventlistener.
...
 		Timestamp:  time.Now(),
 		Data:       updateData,
+		Offset:     &event.Offset,
 	}
 
 	err = p.repos.GameSession.AddGameSessionUpdate(ctx, update)
-	if err != nil {
+	if err != nil && err != gamesessions.ErrUpdateAlreadyProcessed {
 		return err
+	} else {
```
Comment (justefg) [inline eventprocessor/handlers.go:305]: else's needed here

File: migrations/0008_gs_update_offset.up.sql [hunk 2]
```diff
@@ -0,0 +1 @@
+ALTER TABLE game_session_updates ADD COLUMN "offset" NUMERIC UNIQUE;
```
Comment (justefg) [inline migrations/0008_gs_update_offset.up.sql:1]: are quotes necessary?

File: eventprocessor/handlers.go [hunk 3]
```diff
@@ -262,11 +296,16 @@ func onGameMessage(ctx context.Context, p *EventProcessor, event *eventlistener.
...
+		log.Warn().Msgf("Handled already processed event for session: %d, type: %d, offset: %d",
```
Comment (justefg) [inline eventprocessor/handlers.go:306]: else is not needed here

File: eventprocessor/handlers.go [hunk 4]
```diff
@@ -181,21 +199,29 @@ func onGameFinished(ctx context.Context, p *EventProcessor, event *eventlistener
...
```
Comment (justefg) [inline eventprocessor/handlers.go:209]: else is not needed here

File: migrations/0008_gs_update_offset.up.sql (hunk 2 above)
Comment (algys) [reply migrations/0008_gs_update_offset.up.sql:1]: yup, offset is reserved identifier

File: eventprocessor/handlers.go (hunk 3 above)
Comment (algys) [reply eventprocessor/handlers.go]: fixed

File: eventprocessor/handlers.go (hunk 4 above)
Comment (algys) [reply eventprocessor/handlers.go]: fixed

**Review by justefg — APPROVED** [review]

**Pull request merged**